### PR TITLE
Upgrade Storyscript to 0.14.3

### LINKED ISSUE #255: Upgrade Storyscript to 0.14.1
The unit tests in Resolver.py which compile Storyscript lines and calls the resolver to test them fail. This is because Storyscript now checks for variables which are used without any initialisation. Resolver.py used to set variables in the context, and then compile a Storyscript line directly and execute it in a haphazard manner to test the actual Resolver:
https://github.com/asyncy/platform-engine/blob/1c1ca6d7c24dfba3aeccf0ce71df0c81e4a9651b/tests/unit/utils/Resolver.py#L48-L77

However, since then, the engine has a new way of running actual Storyscript as integration tests, with various types of assertions:
https://github.com/asyncy/platform-engine/blob/491fcac1fcd3168e4025877745e111414c6100e5/tests/integration/processing/Lexicon.py#L29

The way these run are explained here:
https://github.com/asyncy/platform-engine/blob/491fcac1fcd3168e4025877745e111414c6100e5/tests/integration/processing/Lexicon.py#L436-L447

### PR DESCRIPTION
**Pull request by anukul**
Closes #255

### CODE REVIEW

**Comment by wilzbach** [comment]
0.14.2 is released by now.

**Comment by anukul** [comment]
Ready for review!

File: tests/unit/utils/Resolver.py [hunk 1]
```diff
@@ -44,50 +199,238 @@ def test_expression_invalid_type():
...
+                          'if !i\n'
+                          '    success = true',
+        cases=[
+            TestCase(assertion=ContextAssertion(key='success', expected=True))
+        ]
+    ),
+    TestSuite(
+        preparation_lines='i = null\n'
+                          'success = true\n'
+                          'if !i\n'
```
Comment (anukul) [inline tests/unit/utils/Resolver.py]: !null is **not** true?

File: tests/unit/utils/Resolver.py [hunk 2]
```diff
@@ -44,50 +199,238 @@ def test_expression_invalid_type():
...
-    [{'i': 5}, 'if i < 5\n  echo foo', False, False],
-    [{'i': 5}, 'if i < 6\n  echo foo', True, False],
-    [{'i': 5}, 'if i <= 5\n  echo foo', True, False],
-    [{'i': 5}, 'if i <= 4\n  echo foo', False, False],
-    [{'i': 5}, 'if i <= 6\n  echo foo', True, False],
-    [{'i': False}, 'if i\n  echo foo', False, False],
-    [{'i': True}, 'if i\n  echo foo', True, False],
-    [{'i': True}, 'if !i\n  echo foo', False, False],
-    [{'i': False}, 'if !i\n  echo foo', True, False],
-    [{'i': None}, 'if !i\n  echo foo', False, False],
```
Comment (anukul) [inline tests/unit/utils/Resolver.py]: !null is **not** true?

Comment (judepereira) [reply tests/unit/utils/Resolver.py]: In this case, null is treated as false.

Comment (anukul) [reply tests/unit/utils/Resolver.py]: So if `null` is `false`, then `!null` should be `true`, right?

**Comment by judepereira** [comment]
@anukul See the Codecov report - it's reporting lesser coverage, when there should've ideally been more.

File: tests/unit/utils/Resolver.py (hunk 2 above)
Comment (judepereira) [reply tests/unit/utils/Resolver.py]: Yes.

**Pull request merged**